### PR TITLE
[Core][FluidDynamicsApplication] Changing from AreaNormal to Normal

### LINKED ISSUE #940: Changing the Normal method behavior
The pull request correcting geometry normal #936 is the first step to change the behavior of Normal method in geometry. In this step the `Normal` changes to `UnitNormal` and defines `AreaNormal` which is not unit length. In a second step the `AreaNormal` will be changed to `Normal`.  This issue is a reminder for the second step.

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #940

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_conditions/embedded_ausas_navier_stokes_wall_condition.h [hunk 1]
```diff
@@ -813,12 +813,12 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) EmbeddedAusasNavierStokesWallCondit
                 face_id,
                 GeometryData::GI_GAUSS_2);
 
-            p_ausas_modified_sh_func->ComputePositiveExteriorFaceAreaNormals(
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_conditions/embedded_ausas_navier_stokes_wall_condition.h]: I would only change what concerns to the geometries. Besides, I consider the old method names to be much more self-explicative.

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
I would only change what concerns to the geometries. Besides, I consider the old method names to be much more self-explicative.

**Review by rubenzorrilla — DISMISSED** [review]
Blocking until we reach consensus.

**Comment by loumalouomega** [comment]
OK

**Comment by loumalouomega** [comment]
@rubenzorrilla  Now it just changes the geometries files and the calls to the Normal methods

File: applications/FluidDynamicsApplication/custom_conditions/embedded_ausas_navier_stokes_wall_condition.h [hunk 2]
```diff
@@ -843,7 +843,7 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) EmbeddedAusasNavierStokesWallCondit
             (rData.area_normals_container).clear();
             for (unsigned int i_gauss = 0; i_gauss < n_gauss; ++i_gauss) {
```
Comment (loumalouomega) [inline applications/FluidDynamicsApplication/custom_conditions/embedded_ausas_navier_stokes_wall_condition.h:845]:
```suggestion
            // We compute the area normal in each Gauss point
            for (unsigned int i_gauss = 0; i_gauss < n_gauss; ++i_gauss) {
```

**Comment by loumalouomega** [comment]
@rubenzorrilla ping

**Comment by rubenzorrilla** [comment]
OK to what concerns the modified shape functions. However this decision was took 1 year ago, so l'd rather wait for the @KratosMultiphysics/technical-committee approval.

**Comment by loumalouomega** [comment]
Ok

**Comment by loumalouomega** [comment]
Ping @KratosMultiphysics/technical-committee

**Review by RiccardoRossi — COMMENTED** [review]
Area Normal has been changed to Normal in geometry?

@rubenzorrilla is this change ok for you? please review this

**Comment by RiccardoRossi** [comment]
removing the FastPR label btw

**Comment by loumalouomega** [comment]
@RiccardoRossi in #940 @pooyan-dadvand  said:

> The pull request correcting geometry normal #936 is the first step to change the behavior of Normal method in geometry. In this step the `Normal` changes to `UnitNormal` and defines `AreaNormal` which is not unit length. In a second step the `AreaNormal` will be changed to `Normal`. This issue is a reminder for the second step.

**Review by RiccardoRossi — COMMENTED** [review]
Are u sure this change will not break existing code?

**Comment by loumalouomega** [comment]
I run all the tests and all checks are green

**Comment by rubenzorrilla** [comment]
@RiccardoRossi did you mean about keeping `AreaNormal()` instead of `Normal()` to return the weighted normal? If this is the case, yes. I'd rather keep `AreaNormal()` which is a much more self-descriptive name than `Normal()`.

Actually, I don't see the problem in having `AreaNormal()` and `UnitNormal()`.

**Comment by jcotela** [comment]
@RiccardoRossi @rubenzorrilla I believe that what @loumalouomega has implemented here matches the goals we set in the corresponding meeting of the committee. The argument for `Normal` instead of `AreaNormal` was to say that the method would implement the "fastest/most convenient possible" way of obtaining a normal vector.

In triangles it is easy to give it a modulus equal to the normal (`0.5 edge1 x edge2`) but the argument was that this is not true in general or for other geometries, where the area would be something like `Sum (modulus_of_normal_on_gauss_point_i * weight_of_gauss_point_i)` (or something of the sort, I don't remember the details). In the end it was decided to just go with `Normal` and not give any assurance that its modulus would be related to the area of the geometry.

Having said that, I do not have a preference for a specific solution, but I think it is fair to say that this is what was originally requested.

**Comment by rubenzorrilla** [comment]
Makes sense. If this was @KratosMultiphysics/technical-committee decision,  I'm ok with `Normal()`.

**Comment by RiccardoRossi** [comment]
@jcotela it is quite important that at least for triangles Normal() (or AreaNormal) delivers the normal multiplies by the Area....

If we give back something equivalent to

     array_1d<double,3> n = ZeroVector(3)
     for(unsigned int igauss = 0; igauss<ngauss; igauss++)
     {
            compute ngauss_unity
            n += integration_weight * ngauss_unity;
     }

so that the the n given back has a norm proportinal to the area of geometry in the direction of the n
than i am ok... if it is not so i need to block since otherwise this would break a lot of things

**Comment by jcotela** [comment]
@RiccardoRossi I fully agree with you, but if I remember correctly, the argument at the time was that the name `AreaNormal` did not reflect this property either. In any case, let me reiterate that I would not be against an eventual change of name, I am just trying to recall what were the arguments for the decisions agreed on the committee meeting.

Also note that I believe that your definition is not consistent with a naive implementation of `AreaNormal` in triangles (since your gauss_weights sum to 0.5, if I remember correctly).

**Comment by loumalouomega** [comment]
Then, what should I do?

**Comment by loumalouomega** [comment]
How to proceed? @KratosMultiphysics/technical-committee

**Comment by roigcarlo** [comment]
forwarding to @KratosMultiphysics/implementation-committee

**Comment by loumalouomega** [comment]
This was the design proposed by @pooyan-dadvand in #940

**Review by AlejandroCornejo — DISMISSED** [review]
approved by the implementation committee but need some documentation

File: kratos/geometries/line_2d_2.h [hunk 3]
```diff
@@ -811,7 +811,7 @@ class Line2D2 : public Geometry<TPointType>
      * point in where the unit normal is to be computed
      * @return The area normal in the given point
```
Comment (adityaghantasala) [inline kratos/geometries/line_2d_2.h]: not area normal ... but normal scaled with the area

**Review by GuillermoCasas — CHANGES_REQUESTED** [review]
It is not the 'normal scaled with the area', but rather the normal (external?) whose modulus is equal to the area of the underlying geometry. Please, could your documentation reflect this, more accurate, description?

**Comment by GuillermoCasas** [comment]
Going back to the comments by @jcotela I realize that even the description I suggest in my previous comment is not accurate. Perhaps a more flexible description of what the method returns would be 'it returns a vector that is normal to its corresponding geometry', since there is no guarantee that it will be easy or feasible (see @jcotela comment) to implement a normal whose modulus is equal to the area in all cases. For a triangle, nonetheless, the previously mentioned description would be ok.

**Comment by loumalouomega** [comment]
> it returns a vector that is normal to its corresponding geometry'

OK, I update the description

**Review by GuillermoCasas — APPROVED** [review]

**Pull request merged**